### ISSUE TITLE
Max pooling layer

### ISSUE THREAD

**Issue by mleming** [jeremykawahara/ann4brains#3]
Hello, I am trying to implement a max pooling layer and a softmax layer into the code. I'm looking at nets.py:

```python
def create_architecture(self, mode, hdf5_data):
        """Returns the architecture (i.e., caffe prototxt) of the model.

        Jer: One day this should probably be written to be more general.
        """

    arch = self.arch
    pars = self.pars
    n = caffe.NetSpec()
    
    if mode == 'deploy':
        n.data = L.DummyData(shape=[dict(dim=pars['deploy_dims'])])
    elif mode == 'train':
        n.data, n.label = L.HDF5Data(batch_size=pars['train_batch_size'], source=hdf5_data, ntop=pars['ntop'])
    else:  # Test.
        n.data, n.label = L.HDF5Data(batch_size=pars['test_batch_size'], source=hdf5_data, ntop=pars['ntop'])
    
    # print(n.to_proto())
    in_layer = n.data
    
    for layer in arch:
        layer_type, vals = layer
    
        if layer_type == 'e2e':
            in_layer = n.e2e = e2e_conv(in_layer, vals['n_filters'], vals['kernel_h'], vals['kernel_w'])
        elif layer_type == 'e2n':
            in_layer = n.e2n = e2n_conv(in_layer, vals['n_filters'], vals['kernel_h'], vals['kernel_w'])
        elif layer_type == 'fc':
            in_layer = n.fc = full_connect(in_layer, vals['n_filters'])
        elif layer_type == 'out':
            n.out = full_connect(in_layer, vals['n_filters'])
            # Rename to user specified unique layer name.
            # n.__setattr__('out', n.new_layer)
    
        elif layer_type == 'dropout':
            in_layer = n.dropout = L.Dropout(in_layer, in_place=True,
                                             dropout_param=dict(dropout_ratio=vals['dropout_ratio']))
        elif layer_type == 'relu':
            in_layer = n.relu = L.ReLU(in_layer, in_place=True,
                                       relu_param=dict(negative_slope=vals['negative_slope']))
        elif layer_type == 'pool':
            in_layer  = L.Pooling(in_layer, kernel_size=3, stride=2, in_place=True, pool=P.Pooling.MAX)
        elif layer_type == 'softmax':
            in_layer = L.SoftmaxWithLoss(in_layer, 1)
        else:
            raise ValueError('Unknown layer type: ' + str(layer_type))

        # ~ end for.

    if mode != 'deploy':
        if self.pars['loss'] == 'EuclideanLoss':
            n.loss = L.EuclideanLoss(n.out, n.label)
        else:
            ValueError("Only 'EuclideanLoss' currently implemented for pars['loss']!")
    return n
```

In particular:

```python
        elif layer_type == 'pool':
            in_layer  = L.Pooling(in_layer, kernel_size=3, stride=2, in_place=True, pool=P.Pooling.MAX)
        elif layer_type == 'softmax':
            in_layer = L.SoftmaxWithLoss(in_layer, 1)
```

I know this is wrong, but how would I properly implement these layers here? The documentation of PyCaffe is quite poor.

**Comment by jeremykawahara** [comment]
I'm not sure sorry - haven't used PyCaffe in a while. You can also manually modify the prototext files and use those (ignore the Python wrapper we created here).
